components/tools/OmeroJava/test/integration/ObjectPropertiesTest.java

```diff
@@ -0,0 +1,632 @@
+package integration;
+
+import java.util.Arrays;
+import java.util.Random;
+
+import omero.ServerError;
+import omero.api.IRenderingSettingsPrx;
+import omero.model.Channel;
+import omero.model.Dataset;
+import omero.model.Folder;
+import omero.model.Image;
+import omero.model.ImportJob;
+import omero.model.ImportJobI;
+import omero.model.JobStatus;
+import omero.model.LogicalChannel;
+import omero.model.Namespace;
+import omero.model.NamespaceI;
+import omero.model.OriginalFile;
+import omero.model.Pixels;
+import omero.model.Plate;
+import omero.model.PlateAcquisition;
+import omero.model.Project;
+import omero.model.Reagent;
+import omero.model.RenderingDef;
+import omero.model.Roi;
+import omero.model.Screen;
+import omero.model.StageLabel;
+import omero.model.TagAnnotation;
+import omero.model.TagAnnotationI;
+import omero.model.Well;
+import omero.sys.ParametersI;
+
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import ome.api.JobHandle;
+
+public class ObjectPropertiesTest extends AbstractServerTest {
+
+    /**
+     * Creates a name with length as a string
+     * @param integer lenght of the name to create in bytes
+     * @return the name
+     * @throws Exception unexpected
+     */
+    private String createName(int integer) throws Exception {
+        final Random rng = new Random();
+        final char[] chars = new char[integer];  // string length
+        for (int i = 0; i < chars.length; i++) {
+            chars[i] = (char) ('a' + rng.nextInt(26));
+        }
+        final String string = new String(chars);
+        return string;
+    }
+
+    /**
+     * Sets up the testing user and group before each test
+     * @throws Exception unexpected
+     */
+    @BeforeMethod
+    private void createUserAndGroup() throws Exception {
+        newUserAndGroup("rw----");
+    }
+
+    /**
+     * Test to create a (tag) annotation and
+     * save it with long name and long namespace
+     * also has a negative test for names and namespaces >2KB
+     * @throws Exception unexpected
+     */
+    @Test
+    public void testAnnotationNameAndNsSaving() throws Exception {
+
+        /* Tag annotation is used here as a good representative
+         * to test the annotations in general */
+        final TagAnnotation ann = new TagAnnotationI();
+
+        /* for annotation name sizes of >2KB the test fails */
+        /* createName() creates name with lenght in approx. bytes */
+        String name = createName(1000000);
+        ann.setName(omero.rtypes.rstring(name));
+        try {
+            iUpdate.saveAndReturnObject(ann);
+            Assert.fail("Hibernate operation: could not insert:..."
```

Comment:
this is a strange failure message -- might be good to choose text that explains to a surprised tester what it means that the test failed?  (e.g., "expected not to be able to insert 1MB tag name into DB" or whatever)